File: backend-node-portifolio/src/api/services/UserServices.ts
import { ERoles } from "@prisma/client";
import { User } from "../domain/models/User";
import { UserRepository } from "../repositories/UserRepository";
import { IUserServices } from "./interfaces/IUserServices";

export class UserServices implements IUserServices{
    userRepository: UserRepository;
    constructor() {
        this.userRepository = new UserRepository()
    }
    async FindByRole(role: ERoles): Promise<User[]> {
        return await this.userRepository.FindByRoles(role);
    }
    async Delete(entity: User): Promise<string> {
        return await this.userRepository.Delete(entity);
    }
    async DeleteById(Id: number): Promise<string> {
        return await this.userRepository.DeleteById(Id);
    }
    async Update(entity: User): Promise<string> {
        return await this.userRepository.Update(entity);
    }
    async Save(entity: User): Promise<string> {
        return await this.userRepository.Save(entity);
    }
    async GetAll(): Promise<User[]> {
        return await this.userRepository.ListAll();
    }
    async GetbyId(id: number): Promise<User> {
        return await this.userRepository.GetbyId(id);
    }

}

File: backend/AuthService/src/AuthService.Application/Services/UserServices.cs
using System.Linq.Expressions;
using AuthService.Application.Interfaces;
using AuthService.Domain.Entities;
using AuthService.Domain.Interfaces;

namespace AuthService.Application.Services
{
    public class UserServices : IUserServices
    {
        public readonly IUserRepository userRepository;
        public UserServices(IUserRepository _userRepository)
        {
            this.userRepository = _userRepository;
        }
        public async Task Delete(User entity)
        {
            await this.userRepository.Delete(entity);
        }

        public async Task<User> FindBy(Expression<Func<User, bool>> predicate)
        {
            return await this.userRepository.FindBy(predicate);
        }

        public async Task<User> GetUserByEmail(string email)
        {
            return await this.userRepository.GetUserByEmail(email);
        }

        public async Task<User> GetById(Guid Id)
        {
            return await this.userRepository.GetById(Id);
        }

        public async Task<List<User>> List()
        {
            return await this.userRepository.List();
        }

        public async Task<List<User>> List(int page)
        {
            return await this.userRepository.List(page);
        }

        public async Task Save(User entity)
        {
            await this.userRepository.Save(entity);
        }

        public async Task Update(User entity)
        {
            await this.userRepository.Update(entity);
        }

        public async Task<PaginatedResult<User>> GetByPagination(int page, string? search, string? role, string? status)
        {
            return await this.userRepository.GetByPagination(page, search, role, status);
        }
    }
}